### ISSUE TITLE
[Feature Request] Add support for eager warmup run to Dynamo

### ISSUE THREAD

**Issue by yf225** [pytorch/pytorch#131934]
### 🚀 The feature, motivation and pitch

This feature is crucial for Traceable FSDP2 rollout, by making sure user doesn't need to modify their training loop to manually run 1st iteration in eager.

@awgu also mentioned that this warmup feature is needed for "delayed scaling in float8 since going into the 1st iteration, amaxes are not initialized yet".

There are other lazy-init use cases that we currently sort of point-fix the support for it, such as optimizer.init_group and nn.LazyModule.

In the implementation, we'll need to be careful to make sure that Compiled Autograd will also do the same number of eager runs before compiling the backward graph.

An attempt is done at https://github.com/pytorch/pytorch/pull/131843.

cc @ezyang @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @amjames
